parsons/action_kit/action_kit.py

```diff
@@ -616,5 +625,40 @@ def _split_tables_no_empties(self, table):
             subset_table = Table(subset)
             if blanks:
                 subset_table.table = subset_table.table.cutout(*blanks)
+            logger.debug(f'Column Upload Blanks: {blanks}')
+            logger.debug(f'Column Upload Columns: {subset_table.columns}')
+            if not set(['user_id', 'email']).intersection(subset_table.columns):
+                logger.warning(
+                    f'Upload will fail without user_id or email. '
+                    f'Rows: {subset_table.num_rows}, Columns: {subset_table.columns}'
+                )
             results.append(subset_table)
         return results
+
+    def collect_upload_errors(self, result_array):
+        """
+        Collect any upload errors as a list of objects from bulk_upload_table 'results' key value.
+        This waits for uploads to complete, so it may take some time if you uploaded a large file.
+        `Args:`
+            result_array: list
+                After receiving a dict back from bulk_upload_table you may want to see if there
+                were any errors in the uploads.  If you call collect_upload_errors()
+        `Returns`:
+            [dict]
```

Comment:
I'm assuming the `error_data` result includes `id` so that it's joinable back to the original results. Might be important to include that here, along with any other relevant details about what we get back from the `uploaderror` endpoint.